README.md

```diff
@@ -142,12 +146,47 @@ Data resiliency is not paramount though as this dataset can be rebuilt upon your
 
 Terraboard currently supports configuration in three different ways:
 
-1. Environment variables
-2. CLI parameters
-3. Configuration file (YAML). A configuration file example can be found in the root directory of this repository.
+1. Environment variables **(only usable for mono provider configuration)**
+2. CLI parameters **(only usable for mono provider configuration)**
+3. Configuration file (YAML). A configuration file example can be found in the root directory of this repository and in the `test/` subdirectory.
+
+**Important: all flags/environment variables related to the providers settings aren't compatible with multi-provider configuration! For that, you must use the Yaml config file to be able to configure multiples buckets/providers.**
 
 The precedence of configurations is as described below.
 
+### Multiple buckets/providers
+
+In order to be able to link to Terraboard multiples buckets or even providers, you must use the Yaml configuration method.
+
+- Set the environment variable **CONFIG_FILE** or the flag **-c** / **--config-file** to a valid Yaml config file.
+
+- In the Yaml file, specify your desired providers configuration. For example with two S3 buckets:
+
+```yaml
+aws:
+  - endpoint: http://minio:9000/
+    region: eu-west-1
+    s3:
+      bucket: test-bucket
+      force-path-style: true
+      file-extension: 
+        - .tfstate
+
+  - endpoint: http://minio:9000/
+    region: eu-west-1
+    s3:
+      bucket: test-bucket2
+      force-path-style: true
+      file-extension: 
+        - .tfstate
+
+```
+
+In the case of AWS, don't forget to set **AWS_ACCESS_KEY_ID** and **AWS_SECRET_ACCESS_KEY** environment variables.
+
+That's it! Terraboard will now fetch these two buckets on DB refresh. You can also mix providers like AWS and Gitlab or anything else.
+You can find a ready-to-use docker example with two *MinIO* buckets in the `test/` sub-folder (just swipe the two terraboard services in the docker-compose file). 
```

Comment:
```suggestion
You can find a ready-to-use Docker example with two *MinIO* buckets in the `test/` sub-folder (just swipe the two terraboard services in the docker-compose file). 
```

Reply:
What do you mean by "swipe the services"?

Reply:
> What do you mean by "swipe the services"?

I added in the docker-compose of `/test`, a Terraboard container configured to use a YAML config file and two MinIO buckets.
However, I commented it in favor of the Terraboard lambda container to avoid having two applications simultaneously.

```yaml
services:
  terraboard-dev:
    build:
      context: ../
      dockerfile: ./Dockerfile
    environment:
      AWS_ACCESS_KEY_ID: root
      AWS_SECRET_ACCESS_KEY: mypassword
      AWS_BUCKET: test-bucket
      AWS_REGION: eu-west-1
      AWS_ENDPOINT: http://minio:9000/
      AWS_FORCE_PATH_STYLE: "true"
      TERRABOARD_LOG_LEVEL: debug
      TERRABOARD_NO_LOCKS: "true"
      TERRABOARD_NO_VERSIONING: "true"
      DB_PASSWORD: mypassword
      DB_SSLMODE: disable
      GODEBUG: netdns=go
    depends_on:
      db:
        condition: service_healthy
      minio:
        condition: service_started
    volumes:
      - ../static:/static:ro
    ports:
      - "8080:8080"

  # terraboard-dev-multi-providers:
  #   build:
  #     context: ../
  #     dockerfile: ./Dockerfile
  #   environment:
  #     AWS_ACCESS_KEY_ID: root
  #     AWS_SECRET_ACCESS_KEY: mypassword
  #     DB_SSLMODE: disable
  #     CONFIG_FILE: config/config.yml
  #     GODEBUG: netdns=go
  #   depends_on:
  #     db:
  #       condition: service_healthy
  #     minio:
  #       condition: service_started
  #   volumes:
  #     - ../static:/static:ro
  #     - ./config.yml:/config/config.yml:ro
  #   ports:
  #     - "8081:8080"
```

I'm not really happy with that, maybe we can split the test sub-folder into multiple config folders each with a docker-compose :thinking:

Reply:
Ah, so you mean "swap" then, not "swipe" :-)

Reply:
Yes :laughing:

Reply:
Yes :laughing: